config/src/main/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurer.java

```diff
@@ -373,6 +375,32 @@ public AuthorizationManagerRequestMatcherRegistry anonymous() {
 			return access(AuthenticatedAuthorizationManager.anonymous());
 		}
 
+		public class AuthorizedUrlVariable {
+			private final String variable;
+
+			public AuthorizedUrlVariable(String variable) {
+				this.variable = variable;
+			}
+
+			public AuthorizationManagerRequestMatcherRegistry equalTo(Function<Authentication, String> function) {
+				return access((auth, requestContext) -> {
+					String value = requestContext.getVariables().get(variable);
+					return new AuthorizationDecision(function.apply(auth.get()).equals(value));
+				});
+			}
+
+			public AuthorizationManagerRequestMatcherRegistry equalTo(String expected) {
+				return access((auth, requestContext) -> {
+					String value = requestContext.getVariables().get(variable);
+					return new AuthorizationDecision(value.equals(expected));
+				});
+			}
+		}
+
+		public AuthorizedUrlVariable hasVariable(String variable) {
```

Comment:
Please add a JavaDoc to explain what this method does along with an example usage.